Source/Scene/Scene.js

```diff
@@ -1851,12 +2012,40 @@ define([
     function updatePrimitives(scene) {
         var frameState = scene._frameState;
 
+        scene._groundPrimitives.update(frameState);
+        scene._primitives.update(frameState);
+
+        var shadowMaps = frameState.shadowMaps;
+        var length = shadowMaps.length;
+        for (var i = 0; i < length; ++i) {
+            var shadowMap = shadowMaps[i];
+            shadowMap.update(frameState);
+
+            if (!shadowMap.outOfView && defined(scene._globe) && scene._globe.castShadows) {
+                var sceneCamera = frameState.camera;
+                var sceneCullingVolume = frameState.cullingVolume;
+                var sceneCommandList = frameState.commandList;
+
+                // Update frame state to render from the light camera
+                frameState.camera = shadowMap.shadowMapCamera;
+                frameState.cullingVolume = shadowMap.shadowMapCullingVolume;
+                frameState.commandList = shadowMap.commandList;
+
+                frameState.commandList.length = 0;
+
+                // Collect terrain commands from the light's POV
+                scene._globe.update(frameState);
```

Comment:
`update` is called twice for the globe, once as usual and once for the shadow map, but it looks like `update` is only called once for primitives (or not?).  Can we add a comment about why?